Calendar/BookingHelpers/BookingGenericFunctions.ps1

```diff
@@ -0,0 +1,132 @@
+﻿function SplitDomainFromEmail {
+    param([string] $email)
+    return [string]$email.Split("@")[1]
+}
+
+function IsConsumerMailbox {
+    param([string]$identity)
+
+    try {
+        $consumerMb = Get-ConsumerMailbox  $identity -ErrorAction Ignore
+        return [boolean]$consumerMb.IsProsumerConsumerMailbox -or $consumerMb.IsMigratedConsumerMailbox -or $consumerMb.IsPremiumConsumerMailbox
+    } catch {
+        return $false #consumer mailbox throws error if domain mailbox
+    }
+}
+
+function IsGetMailboxAvailable {
+    if (Get-Command -Name Get-Mailbox -ErrorAction SilentlyContinue) {
+        Write-Host "Validated connection to Exchange Online..." -ForegroundColor Green
+    } else {
+        Write-Error "Get-Mailbox cmdlet not found. Please validate that you are running this script from an Exchange Management Shell and try again."
+        Write-Host "Look at Import-Module ExchangeOnlineManagement and Connect-ExchangeOnline."
+        exit
+    }
+}
+
+function WriteTestTitle {
+    param([string]$title, [System.Boolean]$Success, [string]$errorMessage, [bool]$writeMessageAlways = $false)
+    Write-Host  ($title.PadRight($script:PadCharsMessage) + " : ") -NoNewline
+    if ($Success) {
+        if ($writeMessageAlways) {
+            Write-GreenCheck
+            Write-Host (" (" + $errorMessage + " )") -ForegroundColor Yellow
+        } else {
+            Write-GreenCheck -NewLine
+        }
+    } else {
+        write-RedX
+        Write-Host (" (" + $errorMessage + " )") -ForegroundColor Red
+    }
+}
+
+function Write-GreenCheck {
```

Comment:
>Write-GreenCheck [](http://example.com/codeflow?start=9&length=16)

Think if you want to have your Functions look like PS cmdlets
i.e. Write-GreenCheck vs WriteGreenCheck